### DISCUSSION TITLE
Unable to run distributed-sweep-example.json

### DISCUSSION THREAD

**Discussion by dalyles** [The-OpenROAD-Project/OpenROAD-flow-scripts#1128] [General]
Hello,

I'm trying to run the Autotuner for the example file <code>distributed-sweep-example.json</code>
using this command 
<code>python3 distributed.py --design gcd --platform sky130hd --config distributed-sweep-example.json tune</code>

I am referencing the "How to use" section of this document: https://openroad-flow-scripts.readthedocs.io/en/latest/user/InstructionsForAutoTuner.html

<img width="1623" alt="image" src="https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/95712444/fcc38ca9-da50-4ae0-a5dd-9182c40bc918">

What I'm trying to understand here is why are there incompatibility issues with the python script? I am using python 3.10.11
For context, Installed Ray using this documentation: https://docs.ray.io/en/latest/ray-overview/installation.html
Under the title "Installing From conda-forge"

<img width="833" alt="image" src="https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/95712444/32d408ff-5e57-44ca-9f61-a781afee1180">

Please advise if this was the right direction in getting Ray installed.
Any help is greatly appreciated.

Thanks!

**Comment by dalyles** [comment]
I dowgraded python to 3.9.16 and the result is more or less the same.

<img width="1625" alt="image" src="https://github.com/The-OpenROAD-Project/OpenROAD-flow-scripts/assets/95712444/d6c2521a-6281-4b4d-87df-0c0d2750be8e">

**Comment by maliberty** [comment]
@vvbandeira can you help?

**Comment by vijayank88** [comment]
@dalyles 
Can you try install AutoTuner using below pip commands:
```
pip3 install -U --user 'ray[default,tune]==1.11.0' ax-platform hyperopt nevergrad optuna pandas
pip3 install -U --user colorama==0.4.4 bayesian-optimization==1.4.0
```

**Reply by dalyles** [reply]
Hello,

When I run the first command I get this message:

<code>ERROR: Could not find a version that satisfies the requirement ray[default,tune]==1.11.0 (from versions: 1.13.0, 2.0.0rc0, 2.0.0rc1, 2.0.0, 2.0.1, 2.1.0, 2.2.0, 2.3.0rc0, 2.3.0, 2.3.1, 2.4.0, 2.5.0)
ERROR: No matching distribution found for ray[default,tune]==1.11.0
</code>

Not sure what is happening here.
Thanks!

**Reply by vvbandeira** [reply]
You can try with `1.13.0`, I think the version we had validated might be too old for pip/conda.

**Comment by dalyles** [comment]
Thank you @vvbandeira 

The flow seems to be working. I wanted to ask another question and know if you've run into this issue before:

<code>~/OpenROAD-flow-scripts/flow/util$ python3 distributed.py --design gcd --platform sky130hd  --config ../designs/sky130hd/gcd/autotuner.json tune</code>

<code>[WARNING 06-12 10:27:23] ax.service.utils.with_db_settings_base: Ax currently requires a sqlalchemy version below 2.0. This will be addressed in a future release. Disabling SQL storage in Ax for now, if you would like to use SQL storage please install Ax with mysql extras via `pip install ax-platform[mysql]`.
2023-06-12 10:27:25,401	INFO services.py:1470 -- View the Ray dashboard at http://127.0.0.1:8265
[2023-06-12 10:28:55,871 E 335860 335860] core_worker.cc:137: Failed to register worker 01000000ffffffffffffffffffffffffffffffffffffffffffffffff to Raylet. IOError: [RayletClient] Unable to register worker with raylet. No such file or directory</code>

**Reply by vvbandeira** [reply]
Maybe try to rollback `sqlalchemy` with `pip install 'sqlalchemy<2.0'`

**Reply by dalyles** [reply]
Thank you, this works for the tuning command.

**Comment by dalyles** [comment]
Now when I'm running the sweep command, I run into this issue: 

<code>~/OpenROAD-flow-scripts/flow/util$ python3 distributed.py --design gcd --platform sky130hd  --config distributed-sweep-example.json sweep</code>

<code>[INFO TUN-0012] Log folder /home/dlyles/OpenROAD-flow-scripts/flow/logs/sky130hd/gcd.
2023-06-12 11:39:45,271	INFO services.py:1470 -- View the Ray dashboard at http://127.0.0.1:8265
[2023-06-12 11:41:15,778 E 633219 633219] core_worker.cc:137: Failed to register worker 01000000ffffffffffffffffffffffffffffffffffffffffffffffff to Raylet. IOError: [RayletClient] Unable to register worker with raylet. No such file or directory</code>

Is this a similar issue? 
Thanks

**Reply by vijayank88** [reply]
Can you get `python3 --version`?

**Reply by dalyles** [reply]
I am using version 3.10.9, this is because I installed ray with pip.
This works successfully for the tune command.

Python 3.9.16 (which I previously used when installing using conda-forge) does not worth for the tune or sweep command
due to compatibility issues.

**Reply by vvbandeira** [reply]
Could you try the latest version of Ray? I found [this](https://github.com/ray-project/ray/issues/21479)
